OOP/assignment2/SurveyApplication.java

```diff
@@ -0,0 +1,233 @@
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class SurveySystem {
+
+	static ArrayList<Participant> participantList = new ArrayList<Participant>();
+	static ArrayList<Question> questionList = new ArrayList<Question>();
+
+	static ArrayList<Question> genrateQuestions() {
+
+		final String COMMA_DELIMITER = ",";
+		final String url = "D://Question/question.csv";
+		ArrayList<Question> questionList = new ArrayList<Question>();
+
+		BufferedReader fileReader = null;
+		try {
+
+			String line = "";
+			fileReader = new BufferedReader(new FileReader(url));// file reading
+			fileReader.readLine();
+			while ((line = fileReader.readLine()) != null) {
+
+				Question question = new Question();
+				String[] tokens = line.split(COMMA_DELIMITER);// separator
+				if (tokens.length > 0) {
+					question.setOptions(tokens[2]);
+					question.setQuestion(tokens[0]);
+					if (tokens[1].equals("Multi Select"))
+						question.type = Question.Type.Multi_Select;
+					else if (tokens[1].equals("Single Select"))
+						question.type = Question.Type.Single_select;
+					else
+						question.type = Question.Type.Text;
+
+				}
+				questionList.add(question);
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			try {
+				fileReader.close();
+			} catch (IOException e) {
+
+				e.printStackTrace();
+			}
+		}
+		return questionList;
+
+	}
+
+	static ArrayList<Participant> inputProcess(ArrayList<Question> questionList) {
+		Scanner scanner = new Scanner(System.in);
+
+		Participant participant = new Participant();
+		ArrayList<String> questionsAnswered = new ArrayList<String>();
+		System.out.println("Enter participant name");
+		String name = scanner.next();
+		participant.setName(name);
+		for (Question question : questionList) {
+
+			System.out.println(question.getQuestion());
+			questionsAnswered.add(scanner.next());
+
+		}
+		participant.setQuestionAnswered(questionsAnswered);
+		participantList.add(participant);
+
+		System.out.println("do u want to continue?y/n");
+		if (scanner.next().equals("y")) {
+			System.out.print("hello");
+
+			return inputProcess(questionList);
```

Comment:
ACTION: Why recursion? for simple iterations we should never use recursion because it is an overhead to the JVM. This could be simple done by a while-loop what u were doing earlier. The recursion is not providing any extra feature.